src/codegen/header.rs

```diff
@@ -56,6 +57,6 @@ impl HeaderEmitter {
 
 impl Default for HeaderEmitter {
     fn default() -> Self {
-        HeaderEmitter::new()
+        HeaderEmitter::new(Targets::X86)
     }
 }
```

Comment:
This is no longer a meaningful default. Remove `Default` implementation.

Reply:
This is required by:
```rust
#[derive(Default)]
pub struct CpuCodeGenerator
```

Reply:
It sounds like deriving `Default` for `CpuCodeGenerator` is a mistake then, since this means `new()` and `default()` give different values. For example, `CpuCodeGenerator::<ArmTarget>::default()` gives you a `CpuCodeGenerator<ArmTarget>` containing a `HeaderEmitter` with `Targets::X86`.

Reply:
If we're parameterizing all of the rest of codegen, perhaps it'd be simpler to just add a type param. for `Tgt`? Monomorph. shouldn't add too much since HeaderEmitter is quite small.

Reply:
@samkaufman I used generics for `emit`. I wanted to make `HeaderEmitter` itself generics but couldn't because of an unused-type-parameter error.

https://github.com/samkaufman/morello/pull/10/commits/a648204d0f433166fa3685c9f55d6f3e75392b65